python/mlc_chat/serve/async_engine.py

```diff
@@ -360,14 +371,231 @@ def _engine_step(self) -> None:
         # Apply the request add and abortion.
         requests_to_add, requests_to_abort = self._request_tracker.get_requests_to_add_and_abort()
         for request in requests_to_add:
-            self.engine.add_request(request)
+            self.background_engine.add_request(request)
         for request_id in requests_to_abort:
-            self.engine.abort_request(request_id)
+            self.background_engine.abort_request(request_id)
 
         # Take an engine step.
-        self.engine.step()
+        self.background_engine.step()
 
     def _abort(self, request_id: str):
         """Internal implementation of request abortion."""
         # Notify the request tracker about the abortion.
         self._request_tracker.abort_request(request_id)
+
+
+class _ThreadedEngine:
+    """The threaded engine keeps running a background request processing
```

Comment:
Can we fold this into AsyncThreadedEngine. and then the call would be self._ffi["add_request"]()

Reply:
Great suggestion! Just updated and now it is completely hidden.